### PR TITLE
Feature/capt 1429 automated claim verifiers config per policy

### PR DESCRIPTION
**Pull request by Reyko**
Refactor `AutomatedChecks::ClaimVerifier`

[CAPT-1429](https://dfedigital.atlassian.net/browse/CAPT-1429)

[CAPT-1429]: https://dfedigital.atlassian.net/browse/CAPT-1429?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/models/automated_checks/claim_verifiers/shared.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  module ClaimVerifiers
+    module Shared
+      VERIFIERS = [
+        AutomatedChecks::ClaimVerifiers::Identity,
+        AutomatedChecks::ClaimVerifiers::Qualifications,
+        AutomatedChecks::ClaimVerifiers::Induction,
+        AutomatedChecks::ClaimVerifiers::CensusSubjectsTaught,
+        AutomatedChecks::ClaimVerifiers::Employment,
+        AutomatedChecks::ClaimVerifiers::StudentLoanAmount
+      ].freeze
```
Comment (your) [inline app/models/automated_checks/claim_verifiers/shared.rb:13]: Could you remove the `Induction` and `StudentLoanAmount` ones from the `Shared` namespace?

These are currently shared as in - they are exeuted always - but there is a clause guard to make sure they run on specific policies only - look [here](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/e5faddada83f55356f0c8d84c04852b68508e08f/app/models/automated_checks/claim_verifiers/induction.rb#L18) and [here](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/e5faddada83f55356f0c8d84c04852b68508e08f/app/models/automated_checks/claim_verifiers/student_loan_amount.rb#L13)

I guess we'd need to move them to a separate namespace, or simply include them individually and in addition to the shared ones?

**Review by your — COMMENTED** [review]
Great stuff - see comment to address currently shared verifiers currently using a guard clause to run on selected policies only

File: app/models/automated_checks/claim_verifiers/shared.rb (hunk 1 above)
Comment (Reyko) [reply app/models/automated_checks/claim_verifiers/shared.rb:11]: Thanks for pointing this out @your refactored 👍 

Attempted to remove the guard clauses but it wasn't possible as at [dqt_report_consumer.rb#L46](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/feature/teacher_id/app/models/automated_checks/dqt_report_consumer.rb#L46) we pass the verifiers as an argument and [admin_qualification_report_upload_spec.rb#L35](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/feature/CAPT-1429-automated-claim-verifiers-config-per-policy/spec/requests/admin_qualification_report_upload_spec.rb#L35) failed with a claim that has a `MathsAndPhysics` Policy.

**Review by slorek — APPROVED** [review]
Looks good!

**Review by your — APPROVED** [review]
Awesome work - just not sure if this should be merged right away on the TID branch or wait until that's on master

**Comment by Reyko** [comment]
> Awesome work - just not sure if this should be merged right away on the TID branch or wait until that's on master

Will wait till we merge TID on Monday as mentioned on Slack 👍

**Pull request merged**